sbt/src/sbt-test/classloader-cache/java-serialization/descendant/src/test/scala/test/ReflectionTest.scala

```diff
@@ -1,12 +1,32 @@
 package test
 
+import java.util.concurrent.{ CountDownLatch, TimeUnit }
+
 import org.scalatest._
 
 class ReflectionTest extends FlatSpec {
-  val foo = new Foo
-  foo.setValue(3)
-  val newFoo = reflection.Reflection.roundTrip(foo)
-  assert(newFoo == foo)
-  assert(System.identityHashCode(newFoo) != System.identityHashCode(foo))
+  val procs = 2
+  val initLatch = new CountDownLatch(procs)
+  val loader = this.getClass.getClassLoader
+  val latch = new CountDownLatch(procs)
+  (1 to procs).foreach { i =>
+    new Thread() {
+      setDaemon(true)
+      start()
+      override def run(): Unit = {
+        initLatch.countDown()
+        initLatch.await(5, TimeUnit.SECONDS)
+        val className = if (i % 2 == 0) "test.Foo" else "test.Bar"
+        loader.loadClass(className)
```

Comment:
`test.Foo` extends `test.Bar` so we should end up concurrently trying to load `test.Foo`. This can cause deadlock if the classloaders are not parallel capable.